### PR TITLE
Fix random.seed() invocation; remove importlib

### PR DESCRIPTION
**Pull request by mjpost**

### CODE REVIEW

File: sotastream/__init__.py [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
 
 sys.dont_write_bytecode = True
 
-from importlib import metadata
 
-__version__ = metadata.version(__package__)
+def get_version():
+    package_dir = os.path.dirname(__file__)
+    root_dir = os.path.abspath(os.path.join(package_dir, '..'))
+    pyproject_path = os.path.join(root_dir, 'pyproject.toml')
```
Comment (thammegowda) [inline sotastream/__init__.py]: Thanks for trying to fix this. 
We seem to have a problem here, -- this assumption of dir structure works when loaded directly from git repository but breaks when installed from pip.  Pip doesnt seem to copy pyproject.toml.

Comment (thammegowda) [reply sotastream/__init__.py]: I fixed this in the latest commit.

**Review by thammegowda — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -58,6 +58,9 @@ sotastream = "sotastream.cli:main"
 requires = ["setuptools", "wheel"]
 build-backend = "setuptools.build_meta"
 
+[tool.setuptools.dynamic]
+version = {file = "sotastream/version.txt"}
+
 [tool.setuptools.packages.find]
```
Comment (mjpost) [inline pyproject.toml:64]: Okay, this is good. Any chance that this dynamic feature can parse `__version__ = "1.0.1"` directly from the package `__init__.py` file? That would maybe be more natural, would simplify the code in the package init file, and would save us creating `version.txt`.

Comment (thammegowda) [reply pyproject.toml:64]: `version = {attr = "sotastream.__version__"}` is supposed to do it. 

Previously, it didnt work on clean environment where requirements were not yet installed.

I tried again now and it seems to work. Maybe the newer pip/setuptools addressed the probjem.. It installs requirements first and then accesses `sotastream.__version__`. 

Please test it and let me know.

Comment (mjpost) [reply pyproject.toml:64]: Works locally, LGTM

**Pull request merged**